specs/eip4844/p2p-interface.md

```diff
@@ -139,45 +155,56 @@ Per `context = compute_fork_digest(fork_version, genesis_validators_root)`:
 | `BELLATRIX_FORK_VERSION` | `bellatrix.SignedBeaconBlock` |
 | `CAPELLA_FORK_VERSION`   | `capella.SignedBeaconBlock`   |
 
-#### BeaconBlockAndBlobsSidecarByRoot v1
+No more than `MAX_REQUEST_BLOCKS_EIP4844` may be requested at a time.
 
-**Protocol ID:** `/eth2/beacon_chain/req/beacon_block_and_blobs_sidecar_by_root/1/`
+#### BlobSidecarsByRoot v1
+
+**Protocol ID:** `/eth2/beacon_chain/req/blob_sidecars_by_root/1/`
 
 Request Content:
 
+```python
+class BlobIdentifier(Container):
+    block_root: Root
+    index: uint64
+```
+
 ```
 (
-  List[Root, MAX_REQUEST_BLOCKS]
+  List[BlobIdentifier, MAX_REQUEST_BLOCKS_EIP4844]
 )
 ```
 
 Response Content:
 
 ```
 (
-  List[SignedBeaconBlockAndBlobsSidecar, MAX_REQUEST_BLOCKS]
+  List[BlobSidecar, MAX_REQUEST_BLOCKS_EIP4844]
```

Comment:
Why don't we want `SignedBlobSidecar` here? Wouldn't we want to verify the proposer's signature too?

Reply:
You already have the block if you are making this request thus you know the proposer signed the block and are looking for it's deterministic dependencies. I don't think you need to transmit (or check) the signature in this case.

The sidecar signature is primarily for the case/race condition in which you received the sidecar before the block which determines it's correctness

Reply:
depending on if we send kzg proofs for singles or only for the aggregate, I could see an argument for keeping the signature. that is, if you have kzg_proofs just for the aggregate, then you can't do a relatively cheap check as you get singles here and need to wait to get them all. Whereas if you get individual proofs, you can verify them each quickly as they come (as a dependency of the block and the valid kzg commitment)

Reply:
> I could see an argument for keeping the signature. that is, if you have kzg_proofs just for the aggregate, then you can't do a relatively cheap check as you get singles here and need to wait to get them all

Yep, that's what I was thinking

Reply:
> depending on if we send kzg proofs for singles or only for the aggregate, I could see an argument for keeping the signature. that is, if you have kzg_proofs just for the aggregate, then you can't do a relatively cheap check as you get singles here and need to wait to get them all. Whereas if you get individual proofs, you can verify them each quickly as they come (as a dependency of the block and the valid kzg commitment)

The plan at the moment is to provide individual KZG proofs as described here https://github.com/ethereum/consensus-specs/pull/3244#discussion_r1104236893 .

Reply:
Since we don't do aggregates anywhere any more, the current construct should be fine, wdyt @terencechain ?

Reply:
Ah, and we also have access to the fast multi-proof verifier.

@asn-d6 what's the performance ratio between verifying a kzg proof and a signature, in general?

Reply:
> Ah, and we also have access to the fast multi-proof verifier.
> 
> @asn-d6 what's the performance ratio between verifying a kzg proof and a signature, in general?

Hmm, I would say both operations have about the same performance. We would need to benchmark them (and run them on the same machine) to get more precise numbers if accuracy is important.

Also both BLS sigs and KZG proofs can be verified in batch.

Reply:
> Since we don't do aggregates anywhere any more, the current construct should be fine, wdyt @terencechain ?

good with me